app/models/operatingsystem.rb

```diff
@@ -238,6 +238,11 @@ def deduce_family
     end
   end
 
+  def self.authorized_resource_name
+    # We don't want STI subclasses to have separate permissions
+    Operatingsystem.name
```

Comment:
Ah, yes. I skipped that one, because that was a module, not class, but the approach would be the same. Nice catch